lib/rudder/analytics/field_parser.rb

```diff
@@ -169,7 +169,8 @@ def parse_common_fields(fields) # rubocop:disable Metrics/AbcSize Metrics/Cyclom
             traits = fields[:traits]
             check_is_hash!(traits, 'traits')
             isoify_dates! traits
-            parsed = parsed.merge({ :traits => traits })
+            # remove top level traits
+            # parsed = parsed.merge({ :traits => traits })
```

Comment:
it just occured to me, to avoind creating a breaking change can we manage this behaviour with a load option like removeTopLevelTraits or something that is false by default?

Reply:
> it just occured to me, to avoind creating a breaking change can we manage this behaviour with a load option like removeTopLevelTraits or something that is false by default?

I think it comes down to whether this would be a breaking change? It is legacy behavior that I understand is no longer supported.  If so, isn't it OK that this is dropped in subsequent SDK versions?  That said, given it's a breaking change, perhaps version the SDK to bump major version?

Reply:
We can do it. But most of our SDKs doesn't have traits on the event level. We can remove it safely with breaking change.

Reply:
@tmchow some developers may use it, if we want to remove it, we have to do it as breaking change.

Reply:
> @tmchow some developers may use it, if we want to remove it, we have to do it as breaking change.

Right but they'd be using an undocumented set of traits that are in the wrong location. This is why I'm suggesting a major version upgrade which indicates a breaking change.

Reply:
Whether it is documented or not documented, they may face data loss.